### ISSUE TITLE
Bug in nx 11.5.0 with Next and babel: regeneratorRuntime is not defined

### ISSUE THREAD

**Issue by cyrus-za** [nrwl/nx#5032]
## Current Behavior

As of nx 11.5.0 and #4944 we can no longer build next apps or serve in prod. This is due to `regeneratorRuntime is not defined` getting thrown, which seems related to the update of .babelrc files. Multiple libs' .babelrc files are being loaded by .nextjs despite only the app's .babelrc file bieng configured for nextjs (the others having react and node babel config). 

## Expected Behavior

#4944 seems to be causing it. 

## Steps to Reproduce

Upgrade any nextjs app from nx 11.x to 11.5.0 using `nx migrate latest` and run the migrations generated. This will result in multiple .babelrc files being updated. After migrating, run `nx run myApp:next-build:production --outputPath=.` where `myApp` is a next.js app. This will throw an error. 

### Failure Logs

Screenshot of vercel build logs
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/1845861/111140592-30c38380-858b-11eb-8171-6aa2f4ef9654.png">

### Environment

This happens when building on vercel, but also happens when github actions run e2e tests (which serves the app in prod mode)

**Comment by huv1k** [comment]
The same error you get when running storybooks.

**Comment by jaysoo** [comment]
We will introduce migrations for this in the next patch version. If you're already on 11.5.1 then the steps above should fix the babel issues.

PR: https://github.com/nrwl/nx/pull/5041

**Comment by cyrus-za** [comment]
Thanks @jaysoo that fixed it for me

**Comment by cyrus-za** [comment]
I spoke too soon. I came from a different branch and forgot to install my node modules when switching. After installing those on the broken branch, I ran into the same issue, despite removing the preset from the root config and adding it to a bunch of non-buildable node libraries.

I still have react libraries that uses `@nrwl/react/babel`
Is this correct? Should they change to next or web?

Why does nextjs log that it is using external babel config for all of those directories. Does nextjs use them only for their directory or does it use the last found babelrc file for the entire project?
